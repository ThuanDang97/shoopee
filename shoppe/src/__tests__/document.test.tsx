// libs
import React from 'react'
import { HtmlContext, HtmlProps } from 'next/dist/shared/lib/html-context'

// component
import Document from '@pages/_document'
import { render } from '@utils/testUtils'

describe('Document render', () => {
  const mockedValue: HtmlProps = {
    __NEXT_DATA__: {
      props: {},
      page: '',
      query: {},
      buildId: '',
    },
    dangerousAsPath: '/',
    docComponentsRendered: {},
    buildManifest: {
      devFiles: [''],
      ampDevFiles: [''],
      polyfillFiles: [''],
      lowPriorityFiles: [''],
      rootMainFiles: [''],
      pages: {
        '/_app': [''],
      },
      ampFirstPages: [''],
    },
    ampPath: '/',
    inAmpMode: false,
    hybridAmp: false,
    isDevelopment: false,
    dynamicImports: [],
    canonicalBase: '',
    headTags: [],
    devOnlyCacheBusterQueryString: '',
    scriptLoader: {},
  }

  it('Should show match Document DOM Snapshot', () => {
    const component = render(
      <HtmlContext.Provider value={mockedValue}>
        <Document />
      </HtmlContext.Provider>,
    )

    expect(component).toMatchSnapshot()
  })
})
