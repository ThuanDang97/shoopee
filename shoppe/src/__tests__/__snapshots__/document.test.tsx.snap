// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document render Should show match Document DOM Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <html
        lang="en"
      >
        <head>
          <meta
            content="0"
            name="next-head-count"
          />
          <script>
            !(function(w,p,f,c){c=w[p]=Object.assign(w[p]||{},{"debug":true});c[f]=(c[f]||[]).concat(["dataLayer.push"])})(window,'partytown','forward');/* Partytown 0.7.2 - MIT builder.io */
!function(t,e,n,i,r,o,a,d,s,c,p,l){function u(){l||(l=1,"/"==(a=(o.lib||"/~partytown/")+(o.debug?"debug/":""))[0]&&(s=e.querySelectorAll('script[type="text/partytown"]'),i!=t?i.dispatchEvent(new CustomEvent("pt1",{detail:t})):(d=setTimeout(w,1e4),e.addEventListener("pt0",f),r?h(1):n.serviceWorker?n.serviceWorker.register(a+(o.swPath||"partytown-sw.js"),{scope:a}).then((function(t){t.active?h():t.installing&&t.installing.addEventListener("statechange",(function(t){"activated"==t.target.state&&h()}))}),console.error):w())))}function h(t){c=e.createElement(t?"script":"iframe"),t||(c.setAttribute("style","display:block;width:0;height:0;border:0;visibility:hidden"),c.setAttribute("aria-hidden",!0)),c.src=a+"partytown-"+(t?"atomics.js?v=0.7.2":"sandbox-sw.html?"+Date.now()),e.body.appendChild(c)}function w(t,n){for(f(),t=0;t&lt;s.length;t++)(n=e.createElement("script")).innerHTML=s[t].innerHTML,e.head.appendChild(n);c&&c.parentNode.removeChild(c)}function f(){clearTimeout(d)}o=t.partytown||{},i==t&&(o.forward||[]).map((function(e){p=t,e.split(".").map((function(e,n,i){p=p[i[n]]=n+1&lt;i.length?"push"==i[n+1]?[]:p[i[n]]||{}:function(){(t._ptf=t._ptf||[]).push(i,arguments)}}))})),"complete"==e.readyState?u():(t.addEventListener("DOMContentLoaded",u),t.addEventListener("load",u))}(window,document,navigator,top,window.crossOriginIsolated);document.currentScript.dataset.partytown="";
          </script>
          <noscript
            data-n-css=""
          />
        </head>
        <body
          class="bg-white"
        >
          <next-js-internal-body-render-target />
          <script
            src="undefined/_next/"
          />
          <script
            id="__NEXT_DATA__"
            type="application/json"
          >
            {"props":{},"page":"","query":{},"buildId":"","scriptLoader":[{"strategy":"worker","src":"https://www.googletagmanager.com/gtag/js?id=G-B67NTM8Z05"}]}
          </script>
        </body>
      </html>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <script
      data-nscript="afterInteractive"
      id="google-tag-manager"
      type="text/partytown"
    >
      
            window.dataLayer = window.dataLayer || [];
            window.gtag = function gtag(){window.dataLayer.push(arguments);}
            gtag('js', new Date());

            gtag('config', 'G-B67NTM8Z05', {
                page_path: window.location.pathname,
            });
        
    </script>
    <div
      class="chakra-portal"
    >
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <html
      lang="en"
    >
      <head>
        <meta
          content="0"
          name="next-head-count"
        />
        <script>
          !(function(w,p,f,c){c=w[p]=Object.assign(w[p]||{},{"debug":true});c[f]=(c[f]||[]).concat(["dataLayer.push"])})(window,'partytown','forward');/* Partytown 0.7.2 - MIT builder.io */
!function(t,e,n,i,r,o,a,d,s,c,p,l){function u(){l||(l=1,"/"==(a=(o.lib||"/~partytown/")+(o.debug?"debug/":""))[0]&&(s=e.querySelectorAll('script[type="text/partytown"]'),i!=t?i.dispatchEvent(new CustomEvent("pt1",{detail:t})):(d=setTimeout(w,1e4),e.addEventListener("pt0",f),r?h(1):n.serviceWorker?n.serviceWorker.register(a+(o.swPath||"partytown-sw.js"),{scope:a}).then((function(t){t.active?h():t.installing&&t.installing.addEventListener("statechange",(function(t){"activated"==t.target.state&&h()}))}),console.error):w())))}function h(t){c=e.createElement(t?"script":"iframe"),t||(c.setAttribute("style","display:block;width:0;height:0;border:0;visibility:hidden"),c.setAttribute("aria-hidden",!0)),c.src=a+"partytown-"+(t?"atomics.js?v=0.7.2":"sandbox-sw.html?"+Date.now()),e.body.appendChild(c)}function w(t,n){for(f(),t=0;t&lt;s.length;t++)(n=e.createElement("script")).innerHTML=s[t].innerHTML,e.head.appendChild(n);c&&c.parentNode.removeChild(c)}function f(){clearTimeout(d)}o=t.partytown||{},i==t&&(o.forward||[]).map((function(e){p=t,e.split(".").map((function(e,n,i){p=p[i[n]]=n+1&lt;i.length?"push"==i[n+1]?[]:p[i[n]]||{}:function(){(t._ptf=t._ptf||[]).push(i,arguments)}}))})),"complete"==e.readyState?u():(t.addEventListener("DOMContentLoaded",u),t.addEventListener("load",u))}(window,document,navigator,top,window.crossOriginIsolated);document.currentScript.dataset.partytown="";
        </script>
        <noscript
          data-n-css=""
        />
      </head>
      <body
        class="bg-white"
      >
        <next-js-internal-body-render-target />
        <script
          src="undefined/_next/"
        />
        <script
          id="__NEXT_DATA__"
          type="application/json"
        >
          {"props":{},"page":"","query":{},"buildId":"","scriptLoader":[{"strategy":"worker","src":"https://www.googletagmanager.com/gtag/js?id=G-B67NTM8Z05"}]}
        </script>
      </body>
    </html>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
